pandas/core/internals.py

```diff
@@ -2388,9 +2388,15 @@ def get_values(self, dtype=None):
         # return object dtype as Timestamps with the zones
         if is_object_dtype(dtype):
             f = lambda x: lib.Timestamp(x, tz=self.values.tz)
-            return lib.map_infer(
+            values = lib.map_infer(
                 self.values.ravel(), f).reshape(self.values.shape)
-        return self.values
+
+            if values.ndim == self.ndim - 1:
```

Comment:
can you explain shortly why reshaping is not wrong in `NonConsolidatableMixIn` (as it is today) ?